Lib/test/datetimetester.py

```diff
@@ -2081,16 +2082,36 @@ def test_isoformat(self):
         self.assertEqual(t.isoformat('T'), "0001-02-03T04:05:01.000123")
         self.assertEqual(t.isoformat(' '), "0001-02-03 04:05:01.000123")
         self.assertEqual(t.isoformat('\x00'), "0001-02-03\x0004:05:01.000123")
+
+        self.assertEqual(t.isoformat(basic=True), "00010203T040501.000123")
+        self.assertEqual(t.isoformat('T', basic=True), "00010203T040501.000123")
+        self.assertEqual(t.isoformat(' ', basic=True), "00010203 040501.000123")
+        self.assertEqual(t.isoformat('\x00', basic=True), "00010203\x00040501.000123")
+
         # bpo-34482: Check that surrogates are handled properly.
         self.assertEqual(t.isoformat('\ud800'),
                          "0001-02-03\ud80004:05:01.000123")
         self.assertEqual(t.isoformat(timespec='hours'), "0001-02-03T04")
+        self.assertEqual(t.isoformat(timespec='hours', basic=True), "00010203T04")
```

Comment:
Maybe but the `basic` tests after the extended tests.

Reply:
I wanted to put them together to see the difference directly rather than mapping lines. 

It's much more explicit like that and easier to debug (you technically just need to remove the separators from the line above)

Reply:
And I see that I did not follow that idea for the above tests. So I'll update the previous block

Reply:
Could you consider aligning the expected output?

```python
self.assertEqual(t.isoformat(timespec='hours'),             "0001-02-03T04")
self.assertEqual(t.isoformat(timespec='hours', basic=True), "00010203T04")
```

That might make this more readable.

Reply:
I could, but the columns might get very long. It was only stressed for a single block so I'm not sure I should do it for everything now.